redis/redis.go

```diff
@@ -19,12 +19,25 @@ func New(config ...Config) *Storage {
 	cfg := configDefault(config...)
 
 	// Create new redis client
-	db := redis.NewClient(&redis.Options{
-		Addr:     fmt.Sprintf("%s:%d", cfg.Host, cfg.Port),
-		DB:       cfg.Database,
-		Username: cfg.Username,
-		Password: cfg.Password,
-	})
+
+	var options *redis.Options
+	var err error
+
+	if cfg.URL != "" {
+		options, err = redis.ParseURL(cfg.URL)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
I believe that we should make it better to be idiomatic this code to compiler.
```suggestion
	if options, err = redis.ParseURL(cfg.URL); cfg.URL != "" && err != nil {
		panic(err)
	}
```

Reply:
Does this not, parse the url if it wasnt specified?

Reply:
Yeah, it's correct logic. it was a little suggestion, not mandatory.

Reply:
no if its correct ill do it, i was just curious if this parses the url before checking if its provided.
Like the order of operations that happen here.
parse url, then check cfg.URL != ""